kubeapi/service.go

```diff
@@ -85,3 +88,103 @@ func UpdateService(clientset *kubernetes.Clientset, svc *v1.Service, namespace s
 	return err
 
 }
+
+//ServicePatchSpec holds the relevant information for making a JSON patch on a service
+type ServicePortPatchSpec struct {
+	Op    string          `json:"op"`
+	Path  string          `json:"path"`
+	Value ServicePortSpec `json:"value"`
+}
+
+type ServicePortSpec struct {
+	Name       string `json:"name"`
+	Port       int    `json:"port"`
+	Protocol   string `json:"protocol"`
+	TargetPort int    `json:"targetPort"`
+}
+
+func PortPatch(name, protocol string, port, targetport int) []ServicePortSpec {
+	portPatch := make([]ServicePortSpec, 1)
+	portPatch[0].Name = name
+	portPatch[0].Port = port
+	portPatch[0].Protocol = protocol
+	portPatch[0].TargetPort = targetport
+
+	return portPatch
+}
+
+//ServicePatchSpec holds the relevant information for making a JSON patch on a service
+type ServiceSelectorPatchSpec struct {
+	Op    string              `json:"op"`
+	Path  string              `json:"path"`
+	Value ServiceSelectorSpec `json:"value"`
+}
+
+type ServiceSelectorSpec struct {
+	Pgcluster string `json:"pg-cluster"`
+	Role      string `json:"role"`
+}
+
+func SelectorPatches(servicename, role string) []ServiceSelectorSpec {
+	selectorPatch := make([]ServiceSelectorSpec, 2)
+	selectorPatch[0].Pgcluster = servicename
+	selectorPatch[0].Role = role
+
+	return selectorPatch
+}
+
+// PatchService performs a JSON patch on a service.
+func PatchServicePort(clientset *kubernetes.Clientset, servicename, namespace, op, jsonpath string, portpatch ServicePortSpec) error {
+	var patchBytes []byte
+	var err error
+
+	patch := make([]ServicePortPatchSpec, 1)
+	patch[0].Op = op
+	patch[0].Path = jsonpath
+	patch[0].Value = portpatch
+
+	patchBytes, err = json.Marshal(patch)
+	if err != nil {
+		log.Error("error in converting patch " + err.Error())
+		return err
+	}
+
+	_, err = clientset.CoreV1().Services(namespace).Patch(servicename, types.JSONPatchType, patchBytes)
+	if err != nil {
+		log.Error(err)
+		log.Error("error patching Service " + servicename)
+	}
+	log.Info("patch Service " + servicename)
+	return err
+}
+
+//PatchService performs a JSON patch on a service.
+func PatchServiceSelector(clientset *kubernetes.Clientset, servicename, namespace, op, jsonpath string, selectorpatch ServiceSelectorSpec) error {
+	var patchBytes []byte
+	var err error
+
+	patch := make([]ServiceSelectorPatchSpec, 1)
+	patch[0].Op = op
+	patch[0].Path = jsonpath
+	patch[0].Value = selectorpatch
+
+	patchBytes, err = json.Marshal(patch)
+	if err != nil {
+		log.Error("error in converting patch " + err.Error())
+		return err
+	}
+
+	return applyServicePatch(clientset, namespace, servicename, patchBytes)
+}
+
+func applyServicePatch(clientset *kubernetes.Clientset, namespace, servicename string, patchBytes []byte) error {
+
+	_, err := clientset.CoreV1().Services(namespace).Patch(servicename, types.JSONPatchType, patchBytes)
+	if err != nil {
+		log.Error(err)
+		log.Error("error patching Service " + servicename)
+	}
```

Comment:
Consider combining the function call and err checking into a single line ere, e.g.

```
if _, err := clientset.CoreV1().Services(namespace).Patch(servicename, types.JSONPatchType, patchBytes); err != nil {
    log.Error(err)
    log.Error("error patching Service " + servicename)
}
```

Reply:
Yeah, that works for me, thanks. I updated that function, and now both PatchServiceSelector and PatchServicePort will use this function going forward as that makes for a cleaner, less repetitive implementation.